activesupport/test/cache/stores/memory_store_test.rb

```diff
@@ -61,6 +61,31 @@ def test_nil_coder_bypasses_mutation_safeguard
     assert_same value, @cache.read("key")
   end
 
+  def test_fetch_logging
+    assert_logs("Cache read: foo") { @cache.fetch("foo") }
+    assert_logs("Cache read: bar:foo") { @cache.fetch("foo", namespace: "bar") }
+  end
+
+  def test_read_logging
+    assert_logs("Cache read: foo") { @cache.read("foo") }
+    assert_logs("Cache read: bar:foo") { @cache.read("foo", namespace: "bar") }
+  end
+
+  def test_write_logging
+    assert_logs("Cache write: foo") { @cache.write("foo", "bar") }
+    assert_logs("Cache write: baz:foo") { @cache.write("foo", "bar", namespace: "baz") }
+  end
+
+  def test_delete_logging
+    assert_logs("Cache delete: foo") { @cache.delete("foo") }
+    assert_logs("Cache delete: bar:foo") { @cache.delete("foo", namespace: "bar") }
+  end
+
+  def test_exist_logging
+    assert_logs("Cache exist?: foo") { @cache.exist?("foo") }
+    assert_logs("Cache exist?: bar:foo") { @cache.exist?("foo", namespace: "bar") }
+  end
```

Comment:
Should these tests go in [`activesupport/test/cache/behaviors/cache_logging_behavior.rb`](https://github.com/rails/rails/blob/10b36bee9e68d9d3cf20da0633c3ee273b3ff301/activesupport/test/cache/behaviors/cache_logging_behavior.rb)?

Reply:
I started with that but it got complicated because `MemCacheStore` has a default random namespace, and `FileStore` keys end up being a path, so it ends up being harder to make concrete assertions. I'll see if I can figure something out with regex or special casing the problematic stores.

Reply:
Done. The regular expressions are a little cursed but the tests themselves don't look too bad 😅

Reply:
I pushed a commit that attempts to simplify the patterns.  After factoring out the `key_pattern` helper, I felt like the `assert_logs_key` helper wasn't really worth the extra indirection, but I could go either way.  Let me know what you think.

Reply:
Looks great! Keeping the complexity inside `FileStoreTest` is really nice